python/kvikio/kvikio/benchmarks/utils.py

```diff
@@ -29,19 +36,28 @@ def pprint_sys_info() -> None:
 
     version = kvikio.cufile_driver.libcufile_version()
     props = kvikio.cufile_driver.properties
-    try:
-        import pynvml
-
-        pynvml.nvmlInit()
-        dev = pynvml.nvmlDeviceGetHandleByIndex(0)
-    except ImportError:
-        gpu_name = "Unknown (install nvidia-ml-py)"
-        mem_total = gpu_name
-        bar1_total = gpu_name
-    else:
-        gpu_name = f"{pynvml.nvmlDeviceGetName(dev)} (dev #0)"
-        mem_total = format_bytes(pynvml.nvmlDeviceGetMemoryInfo(dev).total)
-        bar1_total = format_bytes(pynvml.nvmlDeviceGetBAR1MemoryInfo(dev).bar1Total)
+
+    gpu_name = mem_total = bar1_total = "Unknown (install nvidia-ml-py)"
+    if has_pynvml:
+        dev = None
+        with contextlib.suppress(pynvml.NVMLError):
+            pynvml.nvmlInit()
+            dev = pynvml.nvmlDeviceGetHandleByIndex(0)
+
+        if dev is not None:
+            is_mig = False
+            with contextlib.suppress(pynvml.NVMLError):
+                pynvml.nvmlDeviceGetMigMode(dev)
+                is_mig = True
+
+            if not is_mig:
```

Comment:
Are we sure `nvmlDeviceGetMigMode` exist? Maybe to this:
```suggestion
            try:
                pynvml.nvmlDeviceGetMigMode(dev)
            except pynvml.NVMLError, AttributeError:
                pass
            else:
```

Reply:
Yep the function is [always defined]( https://github.com/gpuopenanalytics/pynvml/blob/82cb5e08f56da105a63ca9fcdeb04335bc90950e/pynvml/nvml.py#L3942-L3948 ). Same is true of `nvidia-ml-py`

Reply:
Fair enough, then I would do:
```suggestion
            try:
                pynvml.nvmlDeviceGetMigMode(dev)
            except pynvml.NVMLError:
                pass
            else:
```

Reply:
So I gave this a try. Unfortunately it appears to skip the memory check on non-MIG machines (like a DGX) as well

Looks like we need this change instead

```suggestion
            try:
                pynvml.nvmlDeviceGetMigMode(dev)
            except pynvml.NVMLError:
```